lib/execution_engine2/sdk/SDKMethodRunner.py

```diff
@@ -59,12 +59,16 @@ class SDKMethodRunner:
     def __init__(
         self,
         config,
-        user_id=None,
-        token=None,
+        user_id: str,
+        token: str,
         job_permission_cache=None,
         admin_permissions_cache=None,
         mongo_util=None,
     ):
+        if not user_id or not user_id.strip():
```

Comment:
@MrCreosote  Will this break rpc client calls if only the token is provided and not the user_id ?

Reply:
I'm assuming here that you're talking about someone using an ee2 SDK client to make calls to the ee2 server. If that's the case, then if the token is provided, the user ID is always provided, as the `*Server.py` class does the token lookup and passes both the `token` and `user_id` into the `*Impl.py` class method via the `ctx` object.

Reply:
It doesn't need to be a compiled client for that matter, the same thing would happen in `curl` or `requests` or whatever.